python/lsst/meas/algorithms/psfDeterminer.py

```diff
@@ -63,6 +72,31 @@ class BasePsfDeterminerTask(pipeBase.Task, metaclass=abc.ABCMeta):
     def __init__(self, config, schema=None, **kwds):
         pipeBase.Task.__init__(self, config=config, **kwds)
 
+    def downsampleCandidates(self, inputCandidateList):
+        """Down-sample candidates from the input candidate list.
+
+        Parameters
+        ----------
+        inputCandidateList : `list` [`lsst.meas.algorithms.PsfCandidate`]
+            Input candidate list.
+
+        Returns
+        -------
+        outputCandidateList : `list` [`lsst.meas.algorithms.PsfCandidate`]
+            Down-selected candidate list.
+        """
+        if len(inputCandidateList) <= self.config.maxCandidates:
+            return inputCandidateList
+
+        rng = np.random.RandomState(seed=self.config.downsampleRandomSeed)
+
+        selection = rng.choice(len(inputCandidateList), size=self.config.maxCandidates, replace=False)
```

Comment:
Random selection is a reasonable choice, but might we want to select the top signal-to-noise candidates? Or try to select sources to give as close to a spatially uniform distribution as possible?

Reply:
Driveby review here. I don't think we'd want to do S/N based selection, but spatially uniform sounds good.

But more generally, with the same fixed seed, are we expecting different selection each time because the `len(inputCandidateList)` will vary from one image to another?

Reply:
Also discussed on slack.  signal-to-noise selection would be bad.  Adding spatial uniformity would be good, but at the moment I think that's a premature optimization (with the default config of 300 there's more than enough that a regular sampling should be fine).

The point of the fixed seed is that if you input the same list of 2000 candidates then you will get the same sampling of 300 candidates every time.  If you input a different list of candidates then you'll get a different sampling, but that's fine because it was different in the first place.

Reply:
OK, I think it should be fine especially if the PSF candidates are ordered randomly to begin with.